### ISSUE TITLE
Broken binary compatibility with libjpeg-turbo 2.0

### ISSUE THREAD

**Issue by afontenot** [mozilla/mozjpeg#329]
Hi, I maintain the [AUR package](https://aur.archlinux.org/packages/mozjpeg/) for mozjpeg. I'm currently seeing some binaries built against libjpeg-turbo (which the official packages use) break against mozjpeg. For example the command heif-convert from [libheif](https://github.com/strukturag/libheif) dies immediately with 

    heif-convert: relocation error: heif-convert: symbol jpeg_write_icc_profile version LIBJPEG_8.0 not defined in file libjpeg.so.8 with link time reference

Building libheif against mozjpeg (instead of libjpeg-turbo) fixes the problem. Is there anything I can do to solve the issue from the mozjpeg packaging end short of backporting all of libjpeg-turbo 2.0? Or do we have to wait for a new release of mozjpeg?

**Comment by kornelski** [comment]
The latest version on master should be based on libjpeg-turbo 2.0. Does that work for you?

**Comment by afontenot** [comment]
It is working, after I figured out the necessary cmake parameters:

    cmake -DWITH_JPEG8=1 -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib .

Anything I can do before the next version is released?

**Comment by kornelski** [comment]
I think the default ABI is V6 on purpose, so if the config switch works for you then there's nothing else to do.

**Comment by afontenot** [comment]
Yep, the problem isn't with the config switch though, because the same switch on the current release (3.3.1) gives me the crash I posted above.

    ./configure --prefix=/usr --with-jpeg8 --mandir=/usr/share/man

Only after I grab the latest master (with the associated switch to cmake) does it work.

**Comment by kornelski** [comment]
The 3.3.1 release is for libjpeg-turbo 1.x, so it's not surprising it's not compatible with 2.0.

**Comment by afontenot** [comment]
In that case, I will recommend switching to mozjpeg-git for people who have problems until the next version is released.

**Comment by afontenot** [comment]
At this point this issue is essentially a duplicate of https://github.com/mozilla/mozjpeg/issues/321 so I'll close this issue.
